Dockerfile

```
@@ -1,65 +1,125 @@
+ARG LIBSSL_DEV_VERSION=3.0.18-1~deb12u1
+# Optional: install Rust toolchain only when needed for native wheels
+ARG INSTALL_RUST_TOOLS=0
 
-# Install dependencies
-RUN pip install --no-cache-dir --upgrade pip setuptools wheel && \
-    pip install --no-cache-dir -r requirements.txt
+WORKDIR /build
 
-# Copy project files
+# Build dependencies for native wheels
+RUN set -eux; \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)